agent/consul/leader_connect_ca.go

```diff
@@ -1451,6 +1451,22 @@ func (c *CAManager) SignCertificate(csr *x509.CertificateRequest, spiffeID conne
 		defer c.caLeafLimiter.csrConcurrencyLimiter.Release()
 	}
 
+	// Append our local CA's intermediate if there is one.
+	inter, err := provider.ActiveIntermediate()
+	if err != nil {
+		return nil, err
+	}
+	root, err := provider.ActiveRoot()
+	if err != nil {
+		return nil, err
+	}
+
+	//check if the intermediate expired before using it to sign
+	err = checkExpired(inter)
+	if err != nil {
+		return nil, err
```

Comment:
We should mention that this error pertains to the intermediate (or it might not be clear which cert is expired)